pkg/nitro_enclaves_device_plugin/device_plugin.go

```diff
@@ -165,27 +187,29 @@ func (*NitroEnclavesDevicePlugin) GetPreferredAllocation(context.Context, *plugi
 // Whenever a Device state change or a Device disappears, ListAndWatch
 // returns the new list
 func (nedp *NitroEnclavesDevicePlugin) ListAndWatch(e *pluginapi.Empty, s pluginapi.DevicePlugin_ListAndWatchServer) error {
-	s.Send(&pluginapi.ListAndWatchResponse{Devices: nedp.dev})
+	err := s.Send(&pluginapi.ListAndWatchResponse{Devices: nedp.dev})
+	if err != nil {
+		return err
+	}
 
 	for {
-		select {
-		case <-nedp.stop:
-			return nil
 
-			//TODO: Implement device health check.
-			/* case d := <-nedp.health:
-			//TODO: Add test case to detect device state.
-			d.Health = pluginapi.Unhealthy
-			s.Send(&pluginapi.ListAndWatchResponse{Devices: nedp.dev})
-			*/
-		}
+		<-nedp.stop
```

Comment:
nit: for loop becomes also redundant now, can be just:
```
<-nedp.stop
return nil
```
and commented implementation makes no much sense if you remove select/case. May be just better to clean it as well keeping more generic TODO (if ever going to do that)